Feature/themability

YuriyVaadin opened the pull request:
Fixes #941 

Extended themable mixin and added shadow parts for theming: row, cell, resize-handle
Created default theme and default functional style module
Removed all stuff related to old themability
Document themable parts (in vaadin-grid.html : row, cell, resize-handle)
Added a new theming example (gold)

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vaadin/vaadin-grid/944)
<!-- Reviewable:end -->

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/vaadin/vaadin-grid?pullRequest=944) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/vaadin/vaadin-grid?pullRequest=944) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/vaadin/vaadin-grid?pullRequest=944) it.</sub>

tomivirkki commented:
Reviewed 7 of 28 files at r1.
Review status: 7 of 28 files reviewed at latest revision, 6 unresolved discussions, some commit checks broke.

---

*[vaadin-grid-row-details-mixin.html, line 18 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-grid/944#-KrF4Eh811ysGgqEVRfT:-KrF4Eh811ysGgqEVRfU:bpmo0we) ([raw file](https://github.com/vaadin/vaadin-grid/blob/739e11bf73621538b0139a4838a072ac8573d8b8/vaadin-grid-row-details-mixin.html#L18)):*
> ```HTML
>   <template>
>     <style>
>       [part~="cell"][detailscell] {
> ```

Should we also drop the separate `detailscell` attribute and merge it to the part like `part="cell details-cell"`?

---

*[vaadin-grid-sorter.html, line 12 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-grid/944#-KrF4f23SrkrfP4yhp9h:-KrF4f23SrkrfP4yhp9i:b-lyv5rx) ([raw file](https://github.com/vaadin/vaadin-grid/blob/3a11c8fd9069a35e3f745110d72326e26a0f6240/vaadin-grid-sorter.html#L12)):*
> ```HTML
> visual feedback, and handlers for sorting the grid data.
> ```

Sorter needs parts also now that we're dropping the mixins

---

*[vaadin-grid-styles.html, line 7 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-grid/944#-KrF3R_vXtecF_uDy1q-:-KrF3R_vXtecF_uDy1q0:blgisdm) ([raw file](https://github.com/vaadin/vaadin-grid/blob/739e11bf73621538b0139a4838a072ac8573d8b8/vaadin-grid-styles.html#L7)):*
> ```HTML
> -->
> 
> <link rel="import" href="../polymer/polymer-element.html">
> ```

Unused import

---

*[vaadin-grid-styles.html, line 144 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-grid/944#-KrF0w3xgJgGofqzQTWy:-KrF0w3xgJgGofqzQTWz:b-ffll1r) ([raw file](https://github.com/vaadin/vaadin-grid/blob/739e11bf73621538b0139a4838a072ac8573d8b8/vaadin-grid-styles.html#L144)):*
> ```HTML
> 
>       /* TODO: Sizer styles, relocate */
>       [part~="outersizer"] {
> ```

This should still target the id

---

*[vaadin-grid-styles.html, line 227 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-grid/944#-KrF5-Vbh9F5gOy-8ym0:-KrF5-Vbh9F5gOy-8ym1:b-n237mz) ([raw file](https://github.com/vaadin/vaadin-grid/blob/739e11bf73621538b0139a4838a072ac8573d8b8/vaadin-grid-styles.html#L227)):*
> ```HTML
> 
>       [part~="cell"][reorder-status="dragging"] {
>         background: var(--primary-color, #000);
> ```

Let's drop the primary color property also

---

*[vaadin-grid.html, line 229 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-grid/944#-KrF2QvpkMbLTKpZEu0R:-KrF2QvpkMbLTKpZEu0S:b-bbtykg) ([raw file](https://github.com/vaadin/vaadin-grid/blob/739e11bf73621538b0139a4838a072ac8573d8b8/vaadin-grid.html#L229)):*
> ```HTML
>         }
>       });
>       if (index % 2 === 1) {
> ```

You could use the _toggleAttribute helper here `this._toggleAttribute('odd', index % 2, row)`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid/944)*
<!-- Sent from Reviewable.io -->

YuriyVaadin commented:
Review status: 6 of 30 files reviewed at latest revision, 6 unresolved discussions.

---

*[vaadin-grid-row-details-mixin.html, line 18 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-grid/944#-KrF4Eh811ysGgqEVRfT:-KrFMDghZxRZ_doaBcRv:b-a2ujfl) ([raw file](https://github.com/vaadin/vaadin-grid/blob/739e11bf73621538b0139a4838a072ac8573d8b8/vaadin-grid-row-details-mixin.html#L18)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

Should we also drop the separate `detailscell` attribute and merge it to the part like `part="cell details-cell"`?
</blockquote></details>

Changed.

---

*[vaadin-grid-sorter.html, line 12 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-grid/944#-KrF4f23SrkrfP4yhp9h:-KrFMF4CJX2pd85EnmpY:brg7xie) ([raw file](https://github.com/vaadin/vaadin-grid/blob/3a11c8fd9069a35e3f745110d72326e26a0f6240/vaadin-grid-sorter.html#L12)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

Sorter needs parts also now that we're dropping the mixins
</blockquote></details>

Parts `sorter-indicators` and `sorter-order` added.

---

*[vaadin-grid-styles.html, line 7 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-grid/944#-KrF3R_vXtecF_uDy1q-:-KrFMNu2gpsCbiosywtz:bss4a2b) ([raw file](https://github.com/vaadin/vaadin-grid/blob/739e11bf73621538b0139a4838a072ac8573d8b8/vaadin-grid-styles.html#L7)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

Unused import
</blockquote></details>

Removed.

---

*[vaadin-grid-styles.html, line 144 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-grid/944#-KrF0w3xgJgGofqzQTWy:-KrFMPDIpuIXW8OgOG_n:b-44p7iz) ([raw file](https://github.com/vaadin/vaadin-grid/blob/739e11bf73621538b0139a4838a072ac8573d8b8/vaadin-grid-styles.html#L144)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

This should still target the id
</blockquote></details>

It is now targeting the id.

---

*[vaadin-grid-styles.html, line 227 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-grid/944#-KrF5-Vbh9F5gOy-8ym0:-KrFMRn1GNc1K3y3aAQT:bu4ephv) ([raw file](https://github.com/vaadin/vaadin-grid/blob/739e11bf73621538b0139a4838a072ac8573d8b8/vaadin-grid-styles.html#L227)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

Let's drop the primary color property also
</blockquote></details>

Dropped.

---

*[vaadin-grid.html, line 229 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-grid/944#-KrF2QvpkMbLTKpZEu0R:-KrFMSmAWH89cLafdwCB:bb3p1fd) ([raw file](https://github.com/vaadin/vaadin-grid/blob/739e11bf73621538b0139a4838a072ac8573d8b8/vaadin-grid.html#L229)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

You could use the _toggleAttribute helper here `this._toggleAttribute('odd', index % 2, row)`
</blockquote></details>

Replaced.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid/944)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 15 of 28 files at r1, 9 of 9 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks broke.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid/944#-:-KrFOCPonD-JGf9RftbP:bnfp4nl)*
<!-- Sent from Reviewable.io -->
